lms/djangoapps/grades/models.py

```diff
@@ -436,4 +442,7 @@ def update_or_create_course_grade(cls, user_id, course_id, **kwargs):
             course_id=course_id,
             defaults=kwargs
         )
+        if grade.letter_grade and not grade.date_passed:
```

Comment:
Actually, I now notice that we have a `passed` property on the `CourseGrade` object.  We should use that instead of assumptions based on the `None` value of the `letter_grade` property.